### ISSUE TITLE
play-silhouette 2.0 fails to discard invalid authenticators 

### ISSUE THREAD

**Issue by HackMate** [mohiva/play-silhouette#406]
Hey, I'm not sure if this projects goal is also to keep maintaining the older play compatible releases, but I thought I'd report this anyways. 

Based on the 2.0 tag in Silhouette.scala line 505 it says:
```scala
...discard(authenticator.extract, Future.successful(pr)).map(r => hr.copy(pr))
```

The last part does not map the changed result correctly, it copies the unchanged result prior to authenticatorService.discard(..). Which means, invalid authenticators like the SessionAuthenticator cannot be removed from the Session.

The end of Silhouette.scala:505 should read:
```
....map(r => hr.copy(r))
```

Any chance for an update of the play 2.3 compatible release? Or will there be no future support for play 2.3?

**Comment by akkie** [comment]
Hi,

I'll see if I can provide an update over the weekend. But note you should update to Silhouette 3.0 because this future is really buggy in version 2.0. Please see https://github.com/mohiva/play-silhouette/pull/323

**Comment by HackMate** [comment]
Hey, 

cool. Just wondering though, is 3.0 compatible with play 2.3? 

Because it has 2.4 dependencies which are not compatible with 2.3 if I'm not mistaken.

**Comment by akkie** [comment]
No, It's only compatible with version 2.4

**Comment by akkie** [comment]
Just released version 2.0.1. It should be available in Maven Central within the next hours.
